samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -1337,6 +1457,100 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
 		link_rsrp_subscribe(true) :
 		link_rsrp_subscribe(false);
 		break;
+	case LINK_CMD_SEARCH:
+		if (link_cmd_args.common_option == LINK_COMMON_READ) {
+
+			ret = lte_lc_periodic_search_get(&search_cfg);
+			if (ret) {
+				mosh_error("Reading search configuration failed: %d", ret);
+			} else {
+				mosh_print("Search configuration:");
+				mosh_print("  Loop: %s", search_cfg.loop ? "True" : "False");
+				mosh_print("  Return to pattern: %u", search_cfg.return_to_pattern);
+				mosh_print("  Band optimization: %u", search_cfg.band_optimization);
+				mosh_print("  Pattern count: %d", search_cfg.pattern_count);
+				for (int i = 0; i < search_cfg.pattern_count; i++) {
+					struct lte_lc_periodic_search_pattern *pattern =
+						&search_cfg.patterns[i];
+					mosh_print("  [%d] Type: %s", i,
+						(pattern->type ==
+						 LTE_LC_PERIODIC_SEARCH_PATTERN_RANGE) ?
+						"Range" : "Table");
+
+					if (pattern->type == LTE_LC_PERIODIC_SEARCH_PATTERN_RANGE) {
+						mosh_print("  [%d] Initial sleep: %u", i,
+							pattern->range.initial_sleep);
+						mosh_print("  [%d] Final sleep: %u", i,
+							pattern->range.final_sleep);
+						mosh_print("  [%d] Time to final sleep: %d", i,
+							pattern->range.time_to_final_sleep);
+						mosh_print("  [%d] Pattern endpoint: %d", i,
+							pattern->range.pattern_end_point);
+					} else if (pattern->type ==
+						   LTE_LC_PERIODIC_SEARCH_PATTERN_TABLE) {
+						mosh_print("  [%d] Value 1: %d", i,
+							pattern->table.val_1);
+						mosh_print("  [%d] Value 2: %d", i,
+							pattern->table.val_2);
+						mosh_print("  [%d] Value 3: %d", i,
+							pattern->table.val_3);
+						mosh_print("  [%d] Value 4: %d", i,
+							pattern->table.val_4);
+						mosh_print("  [%d] Value 5: %d", i,
+							pattern->table.val_5);
+					} else {
+						mosh_error("Unknown pattern");
+					}
+				}
+			}
+		} else if (link_cmd_args.common_option == LINK_COMMON_WRITE) {
+
+			if (!search_cfg_given) {
+				mosh_print(
+					"No --search_cfg option given, "
+					"using default configuration");
+				search_cfg.loop = false;
+				search_cfg.return_to_pattern = 1;
+				search_cfg.band_optimization = 1;
+				search_cfg.pattern_count = 1;
+				search_cfg.patterns[0].type = LTE_LC_PERIODIC_SEARCH_PATTERN_TABLE;
+				search_cfg.patterns[0].table.val_1 = 10;
```

Comment:
I think that these default "magics" are worth of own defines.

Reply:
Changed these numbers to match the configuration of normal power level. A comment added where the numbers have been taken.